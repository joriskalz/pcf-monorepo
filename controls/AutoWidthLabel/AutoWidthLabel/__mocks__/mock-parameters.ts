/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
