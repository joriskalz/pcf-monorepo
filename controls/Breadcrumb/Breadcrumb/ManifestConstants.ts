export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Clickable = 'ItemClickable',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
    SelectedKey = 'SelectedKey',
}
