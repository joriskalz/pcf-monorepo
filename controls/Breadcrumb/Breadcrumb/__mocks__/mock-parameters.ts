/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
