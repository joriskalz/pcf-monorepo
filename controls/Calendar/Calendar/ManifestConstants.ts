export const enum InputEvents {
    SetFocus = 'SetFocus',
}
