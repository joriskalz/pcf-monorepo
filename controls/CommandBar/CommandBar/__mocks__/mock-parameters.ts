/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
