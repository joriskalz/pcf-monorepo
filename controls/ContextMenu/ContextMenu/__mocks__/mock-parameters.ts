/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
