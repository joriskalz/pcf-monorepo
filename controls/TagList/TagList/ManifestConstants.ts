export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Enabled = 'ItemEnabled',
    IconName = 'ItemIconName',
    IconColor = 'ItemIconColor',
    TextColor = 'ItemTextColor',
    BackgroundColor = 'ItemBackgroundColor',
    BorderColor = 'ItemBorderColor',
    IconOnly = 'ItemIconOnly',
    Visible = 'ItemVisible',
}

export const TextAlignmentTypes = {
    '0': 'center',
    '1': 'left',
    '2': 'right',
};
