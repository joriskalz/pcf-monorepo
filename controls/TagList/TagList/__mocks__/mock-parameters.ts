/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        SelectedKey: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        MaxHeight: new MockWholeNumberProperty(),
        FontSize: new MockWholeNumberProperty(),
        ItemHeight: new MockWholeNumberProperty(),
    };
}
