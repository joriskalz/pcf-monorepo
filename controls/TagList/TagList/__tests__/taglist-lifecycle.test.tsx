/* eslint-disable sonarjs/no-identical-functions */
import { mount } from 'enzyme';
import { TagList } from '..';
import { ContextEx } from '../ContextExtended';
import { IInputs } from '../generated/ManifestTypes';
import { ItemColumns } from '../ManifestConstants';
import { MockContext, MockState } from '../__mocks__/mock-context';
import { MockDataSet, MockEntityRecord } from '../__mocks__/mock-datasets';
import { getMockParameters } from '../__mocks__/mock-parameters';

// Since requestAnimationFrame does not exist in the test DOM, mock it
window.requestAnimationFrame = jest.fn().mockImplementation((callback) => {
    callback();
});

describe('CommandBar', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        jest.useFakeTimers();
    });
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === 'DIV') {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
        // reset any jest timers
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        if ((setTimeout as any).mock) {
            jest.runOnlyPendingTimers();
            jest.useRealTimers();
        }
    });

    it('renders', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        component.init(context, notifyOutputChanged);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('renders dummy items when no items configured', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        // Simulate there being no items bound - which causes an error on the parameters
        context.parameters.items.error = true;
        component.init(context, notifyOutputChanged);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('does not cause duplicate key warnings if the ItemKey is duplicated', () => {
        jest.spyOn(console, 'error').mockImplementation();
        const { component, context, notifyOutputChanged } = createComponent();

        context.parameters.items = new MockDataSet([
            new MockEntityRecord('1', {
                [ItemColumns.Key]: 'item1',
                [ItemColumns.DisplayName]: 'Tag 1',
            }),
            new MockEntityRecord('2', {
                [ItemColumns.Key]: 'item1',
                [ItemColumns.DisplayName]: 'Tag 2',
            }),
        ]);
        component.init(context, notifyOutputChanged);

        // Check there is no duplicate key message
        mount(component.updateView(context));

        // We check that there are no 'Warning: Encountered two children with the same key' messages
        expect(console.error).toHaveBeenCalledTimes(0);
    });
});

function createComponent() {
    const component = new TagList();
    const notifyOutputChanged = jest.fn();
    const context = new MockContext<IInputs>(getMockParameters());
    (context as unknown as ContextEx).accessibility.assignedTabIndex = 5;
    const items = [
        new MockEntityRecord('1', {
            [ItemColumns.Key]: 'item1',
            [ItemColumns.DisplayName]: 'Tag 1',
            [ItemColumns.IconName]: 'Open',
            [ItemColumns.IconColor]: 'blue',
        }),
        new MockEntityRecord('2', {
            [ItemColumns.Key]: 'item2',
            [ItemColumns.DisplayName]: 'Tag 2',
            [ItemColumns.IconName]: 'New',
            [ItemColumns.IconColor]: 'blue',
        }),
        new MockEntityRecord('3', {
            [ItemColumns.Key]: 'item3',
            [ItemColumns.DisplayName]: 'Tag 3',
            [ItemColumns.IconName]: 'Save',
            [ItemColumns.IconColor]: 'green',
        }),
    ];
    context.parameters.items = new MockDataSet(items);
    const state = new MockState();
    return { component, context, notifyOutputChanged, state };
}
