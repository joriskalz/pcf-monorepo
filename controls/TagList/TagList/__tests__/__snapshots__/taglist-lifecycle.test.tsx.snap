// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandBar renders 1`] = `
<Memo(CanvasTagList)
  disabled={false}
  height={-1}
  items={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderColor": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "1",
          "values": Object {
            "ItemDisplayName": "Tag 1",
            "ItemIconColor": "blue",
            "ItemIconName": "Open",
            "ItemKey": "item1",
          },
        },
        "enabled": undefined,
        "iconColor": "blue",
        "iconName": "Open",
        "iconOnly": undefined,
        "id": "1",
        "key": "item1",
        "name": "Tag 1",
        "textColor": undefined,
        "visible": undefined,
      },
      Object {
        "backgroundColor": undefined,
        "borderColor": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "2",
          "values": Object {
            "ItemDisplayName": "Tag 2",
            "ItemIconColor": "blue",
            "ItemIconName": "New",
            "ItemKey": "item2",
          },
        },
        "enabled": undefined,
        "iconColor": "blue",
        "iconName": "New",
        "iconOnly": undefined,
        "id": "2",
        "key": "item2",
        "name": "Tag 2",
        "textColor": undefined,
        "visible": undefined,
      },
      Object {
        "backgroundColor": undefined,
        "borderColor": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "3",
          "values": Object {
            "ItemDisplayName": "Tag 3",
            "ItemIconColor": "green",
            "ItemIconName": "Save",
            "ItemKey": "item3",
          },
        },
        "enabled": undefined,
        "iconColor": "green",
        "iconName": "Save",
        "iconOnly": undefined,
        "id": "3",
        "key": "item3",
        "name": "Tag 3",
        "textColor": undefined,
        "visible": undefined,
      },
    ]
  }
  onResize={[Function]}
  tabIndex={5}
  themeJSON={null}
  width={-1}
/>
`;

exports[`CommandBar renders dummy items when no items configured 1`] = `
<Memo(CanvasTagList)
  disabled={false}
  height={-1}
  items={
    Array [
      Object {
        "iconName": "Unknown",
        "id": "1",
        "key": "1",
        "name": "Item 1",
      },
      Object {
        "iconName": "Unknown",
        "id": "2",
        "key": "2",
        "name": "Item 2",
      },
      Object {
        "iconName": "Unknown",
        "id": "3",
        "key": "3",
        "name": "Item 3",
      },
    ]
  }
  onResize={[Function]}
  tabIndex={5}
  themeJSON={null}
  width={-1}
/>
`;
